src/scrapers/base-scraper-with-browser.js

```diff
@@ -119,6 +119,9 @@ class BaseScraperWithBrowser extends BaseScraper {
     }
 
     await this.fillInputs(loginOptions.fields);
+    if (loginOptions.preAction) {
```

Comment:
@brafdlog as this is a pre action that might serve other scenarios, there might be situations where the `loginOptions.fields` will not exist on the dom until competing the pre action phase. consider moving this block above line 121 (above `await this.fillInputs(loginOptions.fields);`)

Reply:
Good point. Fixed